arangod/Aql/IResearchViewExecutor.cpp

```diff
@@ -680,59 +672,140 @@ bool IResearchViewExecutorBase<Impl, Traits>::writeRow(ReadContext& ctx,
+                                                             std::vector<irs::columnstore_reader::values_reader_f> const& storedValuesReaders) {
+  irs::columnstore_reader::values_reader_f reader = storedValuesReaders[index];
+  TRI_ASSERT(reader);
+  irs::bytes_ref value{irs::bytes_ref::NIL}; // column value
+  auto ok = reader(doc->value, value);
+  TRI_ASSERT(ok);
+  storedValue[index] = std::move(value);
+}
+
+template<typename Impl, typename Traits>
+void IResearchViewExecutorBase<Impl, Traits>::pushStoredValues(irs::document const* doc,
+                                                               std::vector<irs::columnstore_reader::values_reader_f> const& storedValuesReaders) {
+  auto const& columnsFieldsRegs = this->_infos.getOutNonMaterializedViewRegs();
+  TRI_ASSERT(!columnsFieldsRegs.empty());
+  auto max = (--columnsFieldsRegs.cend())->first;
+  TRI_ASSERT(max >= IResearchViewNode::SortColumnNumber);
+  auto columnFieldsRegs = columnsFieldsRegs.cbegin();
+  if (IResearchViewNode::SortColumnNumber == columnFieldsRegs->first) {
+    ++max;
+  }
+  auto lastColumn = static_cast<size_t>(max);
+  std::vector<irs::bytes_ref> storedValue(lastColumn + 1);
+  if (IResearchViewNode::SortColumnNumber == columnFieldsRegs->first) {
+    getStoredValue(doc, storedValue, lastColumn, storedValuesReaders);
+    ++columnFieldsRegs;
+  }
+  for (; columnFieldsRegs != columnsFieldsRegs.cend(); ++columnFieldsRegs) {
+    getStoredValue(doc, storedValue, static_cast<size_t>(columnFieldsRegs->first), storedValuesReaders);
+  }
+  this->_indexReadBuffer.pushStoredValue(std::move(storedValue));
+}
+
+template<typename Impl, typename Traits>
+bool IResearchViewExecutorBase<Impl, Traits>::getStoredValuesReaders(irs::sub_reader const& segmentReader,
+                                                                     std::vector<irs::columnstore_reader::values_reader_f>& storedValuesReaders) {
+  auto const& columnsFieldsRegs = this->_infos.getOutNonMaterializedViewRegs();
+  if (!columnsFieldsRegs.empty()) {
+    auto max = (--columnsFieldsRegs.cend())->first;
+    TRI_ASSERT(max >= IResearchViewNode::SortColumnNumber);
+    auto columnFieldsRegs = columnsFieldsRegs.cbegin();
+    if (IResearchViewNode::SortColumnNumber == columnFieldsRegs->first) {
+      ++max;
+    }
+    storedValuesReaders.resize(static_cast<size_t>(max + 1));
+    if (IResearchViewNode::SortColumnNumber == columnFieldsRegs->first) {
+      auto sortReader = ::sortColumn(segmentReader);
+      if (!sortReader) {
+        LOG_TOPIC("bc5bd", WARN, arangodb::iresearch::TOPIC)
+            << "encountered a sub-reader without a sort column while "
+               "executing a query, ignoring";
+        return false;
       }
+      storedValuesReaders[static_cast<size_t>(max)] = std::move(sortReader);
+      ++columnFieldsRegs;
+    }
+    // if stored values exist
+    if (max > 0) {
```

Comment:
Please recheck this condition for case where there is no sort and one stored value (max will be 0)